File: frontend/src/app/dtos/seating-plan/seating-plan-response.dto.ts
export enum SectorType {
  standing = 'standing',
  seating = 'seating',
}

export class SeatingPlanResponse {
  id: number;
  name: string;
  locatedIn: number;
  capacity: number;
  sectors: Sector[];
}

export class Sector {
  id: number;
  name: string;
  color: string;
  type: SectorType;
  rows: Row[];
  capacity: number;
}

export class Row {
  seats: Seat[];
}

export class Seat {
  enabled: boolean;
}

File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/seatingplan/SeatingPlanResponseDto.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.seatingplan;

import at.ac.tuwien.sepm.groupphase.backend.entity.SectorType;
import java.util.List;

public record SeatingPlanResponseDto(
    String name, Long locatedIn, Integer capacity, List<Sector> sectors) {

  public record Sector(
      Long id, String name, String color, Integer capacity, SectorType type, List<Row> rows) {}

  public record Row(List<Seat> seats) {}

  public record Seat(Boolean enabled) {}
}
